File: 30_days_Python_HackerRank_Challenge/RunningTimeComplexity.py
#python 2 3
import math
import sys

def is_prime(n):
    if n <= 1: return False
    if n == 2: return True
    if n % 2 == 0: return False
    
    for i in range(3, int(math.sqrt(n)) + 1, 2):
        if n % i == 0:
            return False
    return True

input_data = sys.stdin.read().split()
if input_data:
    t = int(input_data[0])
    for i in range(1, t + 1):
        n = int(input_data[i])
        if is_prime(n):
            print("Prime")
        else:
            print("Not prime")

File: 30_days_C(modern)_HackerRank_Challenge/RunningTimeComplexity.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>
#include <stdbool.h>


void solve() {
    int n;
    scanf("%d", &n);
    
    if (n <= 1) { printf("Not prime\n"); return; }
    if (n == 2) { printf("Prime\n"); return; }
    if (n % 2 == 0) { printf("Not prime\n"); return; }

    bool isPrime = true;
    for (int i = 3; i <= sqrt(n); i += 2) {
        if (n % i == 0) {
            isPrime = false;
            break;
        }
    }
    printf(isPrime ? "Prime\n" : "Not prime\n");
}

int main() {
    int t;
    scanf("%d", &t);
    while (t--) { solve(); }
    return 0;
}
